### PR TITLE
Web - SG-668 update flows for free 2 person orgs

### PR DESCRIPTION
**Pull request by JaredSnider-Bitwarden**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

If a user in a free org tries to exceed max user seats or max collection count, then prompt the user to upgrade their org via dialog which provides a quick path to the upgrade section of the Billing/Subscription page.  

## Code changes

- **apps/web/src/app/organizations/billing/organization-subscription.component.ts** Added logic to ngOnInit to get "upgrade" flag from query string params and automatically open the upgrade section on load of the Billing/Subscription page 
- **apps/web/src/app/organizations/manage/collections.component.ts** On collection creation method call, check if the user is in a free org and if they have hit the max collections limit already in order to determine if we should show them the upgrade dialog.
- **apps/web/src/app/organizations/manage/org-upgrade-dialog/org-upgrade-dialog.component.html and apps/web/src/app/organizations/manage/org-upgrade-dialog/org-upgrade-dialog.component.ts** Created org upgrade dialog using new `<bit-simple-dialog>` component from the component library. 
- **apps/web/src/app/organizations/manage/people.component.ts and apps/web/src/app/organizations/manage/people.component.html** (1) Refactored component to use organization object as a component property instead of having multiple component properties assigned to properties on the org object. (2) On invite user method call, check if the user is in a free org and if they have hit the max users limit already in order to determine if we should show them the upgrade dialog. 
- **apps/web/src/app/organizations/organization.module.ts** Added OrgUpgradeDialogComponent to declarations.
- **apps/web/src/app/shared/shared.module.ts** Added DialogModule to imports and exports in order to get access to the <bit-simple-dialog>` component  in the `org-upgrade-dialog.component.ts`
- **apps/web/src/locales/en/messages.json** Added english translations for new dialog. 
- **libs/components/src/dialog/dialog.module.ts** Added `IconDirective` to exports so the `bit-dialog-icon` would work in the `org-upgrade-dialog.component.html`
## Screenshots

https://user-images.githubusercontent.com/116684653/203099321-9c5b2930-896e-4823-a147-04072f38887e.mov

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: apps/web/src/app/organizations/manage/collections.component.ts [hunk 1]
```diff
@@ -126,6 +130,24 @@ export class CollectionsComponent implements OnInit {
       return;
     }
 
+    if (
+      !collection &&
+      this.organization.planProductType == ProductType.Free &&
+      this.collections.length == this.organization.maxCollections
```
Comment (differsthecat) [inline apps/web/src/app/organizations/manage/collections.component.ts:136]: Any reason to not use strict equality `=== ` in these conditions?

File: apps/web/src/app/organizations/manage/people.component.ts [hunk 2]
```diff
@@ -233,20 +229,42 @@ export class PeopleComponent
...
 
   async edit(user: OrganizationUserUserDetailsResponse) {
+    // Invite User: Add Flow
+    // Click on user email: Edit Flow
+
+    // User attempting to invite new users in a free org with max users
+    if (
+      !user &&
+      this.organization.planProductType == ProductType.Free &&
+      this.users.length == this.organization.seats
```
Comment (differsthecat) [inline apps/web/src/app/organizations/manage/people.component.ts:246]: Same question as above regarding `===`

File: apps/web/src/app/organizations/billing/organization-subscription.component.ts [hunk 3]
```diff
@@ -75,6 +75,10 @@ export class OrganizationSubscriptionComponent implements OnInit, OnDestroy {
   }
 
   async ngOnInit() {
+    if (this.route.snapshot.queryParamMap.get("upgrade")) {
+      this.changePlan();
+    }
```
Comment (differsthecat) [inline apps/web/src/app/organizations/billing/organization-subscription.component.ts:80]: Why is this separate from the route params handling below?

Comment (JaredSnider-Bitwarden) [reply apps/web/src/app/organizations/billing/organization-subscription.component.ts:80]: Yes, the `this.route.params` only contains route params and not query params. So, I had to have a separate section to retrieve the upgrade query param.

Result for URL: https://localhost:8080/#/organizations/0da98da0-e2b4-46cd-a222-af4d0175cbbf/billing/subscription?upgrade=true
image

File: apps/web/src/app/organizations/manage/people.component.ts (hunk 2 above)
Comment (JaredSnider-Bitwarden) [reply apps/web/src/app/organizations/manage/people.component.ts]: Great catch. I'm not sure why I didn't. Thank you. Fixed.

File: apps/web/src/app/organizations/manage/collections.component.ts (hunk 1 above)
Comment (JaredSnider-Bitwarden) [reply apps/web/src/app/organizations/manage/collections.component.ts]: Great catch. I'm not sure why I didn't. Thank you. Fixed.

File: apps/web/src/app/organizations/billing/organization-subscription.component.ts (hunk 3 above)
Comment (differsthecat) [reply apps/web/src/app/organizations/billing/organization-subscription.component.ts:80]: 👍 Understood, thank you!

**Review by differsthecat — APPROVED** [review]
Looks good @JaredSnider-Bitwarden !

**Comment by JaredSnider-Bitwarden** [comment]
Users (admins) with permission to manage users / collections but not billing now get a better dialog body message: 

image

image

**Review by differsthecat — APPROVED** [review]
New changes look good 🙂

**Review by differsthecat — APPROVED** [review]

**Comment by JaredSnider-Bitwarden** [comment]
Removed `needs-qa` label and merging to master per request from QA.

**Pull request merged**